tests/test_optimization/test_smart_keyword_provider.py

```
@@ -222,9 +225,10 @@ def test_priority_sorting(self):
     def test_max_hints_cap(self):
         p = create_provider()
         candidates = [{"text": f"hint_{i}", "priority": "medium"} for i in range(20)]
-        result = p._format_hints(candidates, max_hints=3, tokens_per_hint=80)
+        result, selected = p._format_hints(candidates, max_hints=3, tokens_per_hint=80)
         lines = [l for l in result.split("\n") if l.startswith("hint_")]
```

Ruff (0.15.14)

[error] Ambiguous variable name: `l`

(E741)